setup.py

```diff
@@ -1,45 +0,0 @@
-from setuptools import setup, find_packages
-
-# Read the contents of your README file
-with open("README.md", "r") as fh:
-    long_description = fh.read()
-
-with open("VERSION.txt", "r") as f:
-    __version__ = f.read().strip()
-
-with open("ucc/_version.py", "w") as f:
-    f.write(f"__version__ = '{__version__}'\n")
-# Read the requirements from the requirements.txt file
-with open("requirements.txt", "r") as req_file:
-    requirements = req_file.read().splitlines()
-
-setup(
-    name="ucc",
-    version=__version__,
-    author="Jordan Sullivan, Misty Wahl, Nate Stemen",
-    author_email="jordan@unitary.foundation",
-    description="Unitary Compiler Collection: A quantum circuit interface and compiler for multiple quantum frameworks",
-    long_description=long_description,
-    long_description_content_type="text/markdown",
-    url="https://github.com/unitaryfund/ucc",  # Repository URL
-    packages=find_packages(),  # Automatically find and include all packages in the project
-    classifiers=[
-        "Programming Language :: Python :: 3",
-        "License :: OSI Approved :: GNU General Public License v3 (GPLv3)",
-        "Operating System :: OS Independent",
-    ],
-    python_requires=">=3.12",  # Python version required
```

Comment:
Let's make sure we carry `python_requires` over to the pyproject.toml: [docs](https://python-poetry.org/docs/pyproject#requires-python).

It seems like a few others may have gotten dropped as well, such as the description and classifiers.

Reply:
Ah good point, I missed that. I don't think I was looking for "python" as a project dependency of a python project. It looks like that does the job! Just wanted to make sure we had it somewhere that python 3.12 is required.

Reply:
Yes, I think this would cover it. But you have a good point @natestemen. I'm realizing this is the "legacy" format of pyproject.toml for poetry. As you note, the current version is `https://packaging.python.org/en/latest/guides/writing-pyproject-toml/` (and can compare to what `poetry new <project>` would create. The new format is what I used in the #209 version with uv. I'll look to rewrite this to use that format.